stores/flatten.go

```diff
@@ -60,6 +61,28 @@ func Flatten(in map[string]interface{}) map[string]interface{} {
 	return newMap
 }
 
+func Flatten(md Metadata) (map[string]interface{}, error) {
+	var mdMap map[string]interface{}
+	jsonBytes, err := json.Marshal(md)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(jsonBytes, &mdMap)
+	if err != nil {
+		return nil, err
+	}
+
+	flat := flattenMap(mdMap)
+	macOnlyEncryptedToString(flat)
+	for k, v := range flat {
+		if s, ok := v.(string); ok {
+			flat[k] = strings.Replace(s, "\n", "\\n", -1)
+		}
+	}
```

Comment:
IMO this function (and its unflattening correspond) should be split up into three different functions:
1. a function that converts a metadata object to a flattened map;
2. a function which converts known boolean (or other non-string fields) to strings;
3. a function which replaces `\n` by `\\n`.

Then the first function can be optionally combined - or not combined - with the other two in case of potential other (future) file formats that only need some of these transformations.

Reply:
method names... :thinking: 
1. FlattenMetadata / UnflattenMetadata
2. encodeNonStrings / decodeNonStrings  :thinking:  packNonStrings / unpackNonStrings
3. encodeNewLines / decodeNewLines  .. pack__ / unpack__  

Any opinions?  I'll start with encode__/decode__ for now.  Easy to refactor if we want something different.